pkgs/tools/networking/ppp/default.nix

```diff
@@ -53,6 +53,14 @@ stdenv.mkDerivation rec {
 
   NIX_LDFLAGS = "-lcrypt";
 
+  # This can probably be removed if version > 2.4.9, as IPX support
+  # has been removed upstream[1].  Just check whether pkgsMusl.ppp
+  # still builds.
+  #
+  # [1]: https://github.com/ppp-project/ppp/commit/c2881a6b71a36d28a89166e82820dc5e711fd775
+  NIX_CFLAGS_COMPILE =
+    lib.optionalString stdenv.hostPlatform.isMusl "-UIPX_CHANGE";
```

Comment:
I don't have much context here. Just trying to understand.
Where does `-UIPX_CHANGE` come from?

Reply:
IPX support is only built if `IPX_CHANGE` is defined:

https://github.com/ppp-project/ppp/blob/ppp-2.4.9/pppd/ipxcp.c#L43

But `IPX_CHANGE` is defined unconditionally:

https://github.com/ppp-project/ppp/blob/ppp-2.4.9/pppd/Makefile.linux#L94-L96

So presumably what you're supposed to do is edit that file and remove the `-DIPX_CHANGE` if you want IPX disabled, but for us it's much easier to just use `NIX_CFLAGS_COMPILE` to force undefine `IPX_CHANGE`.  Usually, I might be tempted to submit an upstream PR to make this configurable, but since the feature is entirely removed in HEAD already, there's no point, and something slightly hackish like this is fine.

Reply:
Thanks for the comprehensive explanation.